pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2848,6 +2848,20 @@ public <T> void registerConfigurationListener(String configKey, Consumer<T> list
         configRegisteredListeners.put(configKey, listener);
     }
 
+    /**
+     * Allows the third-party plugin to add a custom dynamic configuration.
+     * <p>
+     * The custom dynamic configuration doesn't support setting up a listener.
+     */
+    public void addCustomDynamicConfiguration(String key, String defaultValue, Predicate<String> validator) {
```

Comment:
This new API is dangerous because some config's dynamic update is not supported now, such as `messageExpiryCheckIntervalInMinutes`,  `serverPort`, the field can be marked `dynamic update` only when we need to have truly implemented the dynamic update.

Reply:
@codelipenghui @Technoboy-  Could you take a look?